azure-iot-device/azure/iot/device/iothub/pipeline/mqtt_topic_iothub.py

```diff
@@ -252,3 +252,38 @@ def encode_properties(message_to_send, topic):
         topic += user_properties_encoded
 
     return topic
+
+
+def get_twin_response_topic_for_subscribe():
```

Comment:
Can we organize these in with the like functions related to other operations?

e.g. `get_<thing>_topic_for_subscribe()` functions are all grouped together, `is_<thing>_topic()` functions are all grouped together, etc.

Reply:
that's a great idea.  I want to revisit this file and put it under test in a separate PR.  Its reached what a friend of mine would call critical mess.  It's sufficiently complex that it needs its own tests.  Plus, this manual URI parsing needs to be replaced by library calls -- splitting strings is just too error-prone.  

---
In reply to: [288783848](https://github.com/Azure/azure-iot-sdk-python-preview/pull/96#discussion_r288783848) [](ancestors = 288783848)